python/kubeflow/trainer/constants/constants.py

```diff
@@ -120,27 +124,49 @@
 # The default PIP index URL to download Python packages.
 DEFAULT_PIP_INDEX_URL = os.getenv("DEFAULT_PIP_INDEX_URL", "https://pypi.org/simple")
 
-# The default command for the Custom Trainer.
-DEFAULT_CUSTOM_COMMAND = ["bash", "-c"]
-
-# The default command for the TorchTune Trainer.
-DEFAULT_TORCHTUNE_COMMAND = ["tune", "run"]
-
-# The default entrypoint for torchrun.
-TORCH_ENTRYPOINT = "torchrun"
+# The exec script to embed training function into container command.
+# __ENTRYPOINT__ depends on the MLPolicy, func_code and func_file is substituted in the `train` API.
+EXEC_FUNC_SCRIPT = textwrap.dedent(
+    """
+        read -r -d '' SCRIPT << EOM\n
+        {func_code}
+        EOM
+        printf "%s" \"$SCRIPT\" > \"{func_file}\"
+        __ENTRYPOINT__ \"{func_file}\""""
+)
```

Comment:
For runtimes with CustomTrainer, I insert the complete `command` when Runtime object is created: 

https://github.com/kubeflow/sdk/blob/f3e90202b9af447afd30e03fcace9f4b842c16cb/python/kubeflow/trainer/utils/utils.py#L155-L164

I think, this is much clear approach since we always have the final  container `command` in `runtime.trainer.__command`.